app/forms/payment_summary_form.rb

```diff
@@ -1,17 +1,38 @@
 class PaymentSummaryForm < BaseForm
-  # TODO: Define accessible attributes, eg attr_accessor :field
+  attr_accessor :temp_payment_method, :type_change, :registration_cards, :registration_card_charge, :total_charge
 
   def initialize(transient_registration)
     super
-    # TODO: Define params to get from transient_registration, eg self.field = @transient_registration.field
+    self.temp_payment_method = @transient_registration.temp_payment_method
+
+    self.type_change = @transient_registration.registration_type_changed?
+    # TODO: Uncomment this once https://github.com/DEFRA/waste-carriers-renewals/pull/139/ is merged in
+    # self.registration_cards = @transient_registration.temp_cards || 0
+    self.registration_cards = 2 # Temp value for testing
+    self.registration_card_charge = determine_registration_card_charge
+    self.total_charge = determine_total_charge
   end
 
   def submit(params)
     # Assign the params for validation and pass them to the BaseForm method for updating
-    # TODO: Define allowed params, eg self.field = params[:field]
-    # TODO: Include attributes to update in the attributes hash, eg { field: field }
-    attributes = {}
+    self.temp_payment_method = params[:temp_payment_method]
+    attributes = { temp_payment_method: temp_payment_method }
 
     super(attributes, params[:reg_identifier])
   end
+
+  validates :temp_payment_method, inclusion: { in: %w[card bank_transfer] }
+
+  private
+
+  def determine_total_charge
+    charges = [Rails.configuration.renewal_charge]
+    charges << Rails.configuration.type_change_charge if type_change
+    charges << registration_card_charge
+    charges.sum
+  end
+
+  def determine_registration_card_charge
+    registration_cards * 5
```

Comment:
To keep things consistent I'd go with making this a config value rather than a magic number.